src/main/java/co/cask/http/PatternPathRouterWithGroups.java

```diff
@@ -107,7 +109,7 @@ public void add(final String source, final T destination) {
    */
   public List<RoutableDestination<T>> getDestinations(String path) {
 
-    String cleanPath = (path.endsWith("/") && path.length() > 1)
+    String cleanPath = (path.endsWith("/") && path.length() > 0)
```

Comment:
This change is not required if split is changed to return trailing empty strings.